frontend/src/components/Dislike/Dislike.jsx

```diff
@@ -0,0 +1,45 @@
+import "./Dislike.css";
+import axios from "axios";
+import { UserContext } from "../App/App";
+import { useContext } from "react";
+
+export default function Dislike({ ticker, stockData, setStockData }) {
+    const { setError } = useContext(UserContext);
+
+    async function handleDislike() {
+        const url = stockData.disliking
+            ? "http://localhost:3000/undislike"
+            : "http://localhost:3000/dislike";
+        try {
+            const response = await axios.post(
+                url,
+                { ticker },
+                { withCredentials: true, validateStatus: () => true }
+            );
+
+            if (response.status === 200) {
```

Comment:
This logic is confusing. Ternaries are not usually used on the left side of making assignments/mutations.

It would be better if the response would bring the actual liking and disliking from the database and:

```
setStockData({
  ...stockData,
  liking: // some value of logic
  disliking: // some value of logic
});

```

Reply:
Should I make an endpoint for checking if a user is currently liking the stock? Wouldn't it make it slower since we have to call the database?

Reply:
No, that's not what I'm suggesting. You are already making a call to "Dislike" or "Undislike". The response payload can come back with the values of liking and disliking already.

In any case, the important thing to do is to not use the ternary on the left side of the assignment. 

NOT
```
  stockData.liking && !stockData.disliking ?
     setStockData({ ...stockData, disliking: !stockData.disliking, liking: false }) :
     setStockData({ ...stockData, disliking: !stockData.disliking });
```

YES
```
 const disliking = // logic here to decide what the value of disliking should be
 const liking = // logic here to decide what the value of liking should be
  setStockData({
    ...stockData
    disliking: 
  }
```